### PR TITLE
[console-Iot] Integrate add device using json with mock server

### PR DESCRIPTION
**Pull request by drcoolsanjeev**
### Type of change

- Enhancement 

### Description

Integrate create device flow with JSON with server
### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: console/console-init/ui/src/modules/iot-device/dialogs/CreateDevice/AddDeviceUsingJsonPage.tsx [hunk 1]
```diff
@@ -39,8 +44,19 @@ export default function AddDeviceUsingJsonPage() {
...
     //TODO: Add query to save iot device
+    if (deviceDetail) {
+      const device: ICreateDeviceRequest = getDeviceFromDeviceString(
+        deviceDetail
+      );
+      const variable = {
+        iotproject: { name: projectname, namespace },
+        device: device
+      };
+      console.log(variable);
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device/dialogs/CreateDevice/AddDeviceUsingJsonPage.tsx:57]: please remove console.log

File: console/console-init/ui/src/modules/iot-device/dialogs/CreateDevice/AddDeviceUsingJsonPage.tsx [hunk 2]
```diff
@@ -39,8 +44,19 @@ export default function AddDeviceUsingJsonPage() {
...
+      await setCreateDeviceQueryVariables(variable);
+    }
     history.push(deviceListRouteLink);
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device/dialogs/CreateDevice/AddDeviceUsingJsonPage.tsx:60]: I think you should call change route code in useMutation query as a callback function based on success or failure.

File: console/console-init/ui/src/modules/iot-device/utils/utils.ts [hunk 3]
```diff
@@ -184,6 +185,65 @@ const getFormInitialStateByProperty = (
   return initialState;
 };
 
+const getDeviceFromDeviceString = (device: string) => {
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device/utils/utils.ts:188]: Please change the name of function something like serializeDevice etc.
Improve logic of this function as we had discussed in call. Don't need try catch here.

Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/iot-device/utils/utils.ts:188]: WIll improve this logic in my next PR once your PR get merged to master

**Pull request merged**